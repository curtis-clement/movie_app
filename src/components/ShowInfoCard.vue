<script setup lang="ts">
import type { ShowInfoCardData } from '@/models/model';

const emit = defineEmits(['showDetails']);

const props = defineProps<{
  actionBannerText: string;
  show: ShowInfoCardData;
}>();
</script>

<template>
  <div class="show-info-card">
    <div class="image-container">
      <img
        v-if="props.show.image"
        :src="props.show.image.medium"
        :alt="props.show.name"
        class="card-image"
      />
    </div>
    <div class="show-info-card-content">
      <h3 class="show-title">{{ props.show.name }}</h3>
      <div class="show-detail"><b>Rating:</b> {{ props.show.rating ? props.show.rating : 'N/A' }}</div>
      <div class="show-detail"><b>Status:</b> {{ props.show.status }}</div>
      <div class="show-detail"><b>Network:</b> {{ props.show.network ? props.show.network.name : 'N/A' }}</div>
      <div class="show-detail"><b>Genres:</b> {{ props.show.genres.join(' / ') }}</div>
    </div>
    <div class="action-banner" @click="emit('showDetails', props.show.id)">
      {{ props.actionBannerText }}
    </div>
  </div>
</template>

<style scoped>
.show-info-card {
  background: white;
  border-radius: 8px;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1);
  display: flex;
  flex-direction: column;
  height: 100%;
  overflow: hidden;
  transition: transform 0.2s;
  width: 100%;
}

.show-info-card:hover {
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.15);
  transform: translateY(-4px);
}

.image-container {
  aspect-ratio: 3/4;
  overflow: hidden;
  width: 100%;
}

.card-image {
  height: 100%;
  object-fit: cover;
  width: 100%;
}

.show-info-card-content {
  flex-grow: 1;
  padding: 1rem;
}

.show-title {
  font-size: 1.25rem;
  font-weight: 600;
  margin: 0 0 0.5rem 0;
}

.show-detail {
  color: #666;
  font-size: 0.875rem;
  margin: 0.5rem 0;
}

.action-banner {
  background-color: #3498db;
  color: white;
  cursor: pointer;
  font-size: 0.875rem;
  margin-top: auto;
  padding: 0.75rem;
  text-align: center;
  transition: background-color 0.2s;
}

.action-banner:hover {
  background-color: #2980b9;
}
</style>