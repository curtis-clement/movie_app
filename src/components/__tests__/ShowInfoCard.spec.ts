import { beforeEach, describe, it, expect } from 'vitest';
import { mount, VueWrapper } from '@vue/test-utils';
import ShowInfoCard from '@/components/ShowInfoCard.vue';
import type { ShowInfoCardData } from '@/models/model';

const mockShow = {
  id: 1,
  name: 'Test Show',
  image: {
    medium: 'http://example.com/image.jpg',
    original: 'http://example.com/image.jpg'
  },
  rating: 8.5,
  status: 'Running',
  network: {
    id: 1,
    name: 'Test Network',
    country: {
      code: 'US',
      name: 'United States',
      timezone: 'America/New_York'
    }
  },
  genres: ['Drama', 'Comedy']
}

const mountComponent = (props: { actionBannerText: string, show: ShowInfoCardData }) => {
  return mount(ShowInfoCard, {
    props
  })
}

describe('ShowInfoCard', () => {
  let wrapper: VueWrapper<InstanceType<typeof ShowInfoCard>>;

  beforeEach(() => {
    wrapper = mountComponent({
      actionBannerText: 'View show details',
      show: mockShow
    });
  });
  
  it('renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('passes props correctly', () => {
    expect(wrapper.props()).toEqual({
      actionBannerText: 'View show details',
      show: mockShow
    });
  });

  it('displays the show image', () => {
    expect(wrapper.find('img').exists()).toBe(true);
    expect(wrapper.find('img').attributes('src')).toBe(mockShow.image.medium);
  });

  it('displays the show details correctly', () => {
    expect(wrapper.text()).toContain(mockShow.name);
    expect(wrapper.text()).toContain(mockShow.rating.toString());
    expect(wrapper.text()).toContain(mockShow.status);
    expect(wrapper.text()).toContain(mockShow.network.name);
    expect(wrapper.text()).toContain(mockShow.genres.join(' / '));
  });
  
  it('emits show-details event when action banner is clicked', () => {
    wrapper.find('.action-banner').trigger('click');
    expect(wrapper.emitted('show-details')).toBeDefined();
    expect(wrapper.emitted('show-details')?.[0]).toEqual([mockShow.id]);
  });
});
  
