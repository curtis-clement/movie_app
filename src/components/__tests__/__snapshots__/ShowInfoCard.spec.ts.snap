// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShowInfoCard > renders the component 1`] = `
"<div data-v-aaef1f9b="" class="show-info-card">
  <div data-v-aaef1f9b="" class="image-container"><img data-v-aaef1f9b="" src="http://example.com/image.jpg" alt="Test Show" class="card-image"></div>
  <div data-v-aaef1f9b="" class="show-info-card-content">
    <h3 data-v-aaef1f9b="" class="show-title">Test Show</h3>
    <div data-v-aaef1f9b="" class="show-detail"><b data-v-aaef1f9b="">Rating:</b> 8.5</div>
    <div data-v-aaef1f9b="" class="show-detail"><b data-v-aaef1f9b="">Status:</b> Running</div>
    <div data-v-aaef1f9b="" class="show-detail"><b data-v-aaef1f9b="">Network:</b> Test Network</div>
    <div data-v-aaef1f9b="" class="show-detail"><b data-v-aaef1f9b="">Genres:</b> Drama / Comedy</div>
  </div>
  <div data-v-aaef1f9b="" class="action-banner">View show details</div>
</div>"
`;
